src/main/java/ru/ralsei/whatcanyousee/gameactivity/GooglePlayHandler.java

```diff
@@ -0,0 +1,641 @@
+                        @Override
+                        public void onSuccess(Player player) {
+                            mPlayerId = player.getPlayerId();
+
+                            activity.getUIHandler().switchToMainScreen();
+                        }
+                    })
+                    .addOnFailureListener(activity.createFailureListener("There was a problem getting the player id!"));
+        }
+
+        mInvitationsClient.registerInvitationCallback(mInvitationCallback);
+
+        GamesClient gamesClient = Games.getGamesClient(activity, googleSignInAccount);
+        gamesClient.getActivationHint()
+                .addOnSuccessListener(new OnSuccessListener<Bundle>() {
+                    @Override
+                    public void onSuccess(Bundle hint) {
+                        if (hint != null) {
+                            Invitation invitation =
+                                    hint.getParcelable(Multiplayer.EXTRA_INVITATION);
+
+                            if (invitation != null && invitation.getInvitationId() != null) {
+                                // retrieve and cache the invitation ID
+                                Log.d(TAG, "onConnected: connection hint has a room invite!");
+                                acceptInviteToRoom(invitation.getInvitationId());
+                            }
+                        }
+                    }
+                })
+                .addOnFailureListener(activity.createFailureListener("There was a problem getting the activation hint!"));
+    }
+
+    /**
+     * Calls after disconnecting from the google play account (etc after internet connection lost).
+     */
+    void onDisconnected() {
+        Log.d(TAG, "onDisconnected()");
+
+        mRealTimeMultiplayerClient = null;
+        mInvitationsClient = null;
+
+        mAchievementsClient = null;
+        mLeaderboardsClient = null;
+
+        activity.clearAllResources();
+        activity.setContentView(R.layout.activity_game);
+        activity.setupListeners();
+        activity.getUIHandler().switchToScreen(R.id.screen_sign_in);
+    }
+
+    /**
+     * Called when connected to the room.
+     */
+    private RoomStatusUpdateCallback mRoomStatusUpdateCallback = new RoomStatusUpdateCallback() {
+        @Override
+        public void onConnectedToRoom(Room room) {
+            Log.d(TAG, "onConnectedToRoom.");
+
+            mMyId = room.getParticipantId(mPlayerId);
+
+            if (mRoomId == null) {
+                mRoomId = room.getRoomId();
+            }
+
+            Log.d(TAG, "Room ID: " + mRoomId);
+            Log.d(TAG, "My ID " + mMyId);
+            Log.d(TAG, "<< CONNECTED TO ROOM>>");
+        }
+
+        @Override
+        public void onDisconnectedFromRoom(Room room) {
+            activity.runOnUiThread(new Runnable() {
+                @Override
+                public void run() {
+                    mRoomId = null;
+                    mRoomConfig = null;
```

Comment:
It's true that roomId and roomConfig could be safely assigned in the callback (afaik in android it's guaranteed there will be no problems with shared data). However, this runnable has to be on UI thread because following code showing a game error (on the screen), so whatever.